packages/lib/src/classes/spot.ts

```diff
@@ -36,87 +36,59 @@ export class Spot {
-                    );
-                    throw new Error(
-                        `Error then retrieving responses from Beam. Abborting requests ...`,
-                    );
-                }
+            console.log(`Created new Beam Task with id ${this.currentTask}`)
 
-                const beamResponseData: Array<BeamResult> =
-                    await beamResponses.json();
+            let eventSource = new EventSource(`${this.url.toString()}beam/${this.currentTask}?wait_count=${this.sites.length}`)
+            eventSource.addEventListener("new_result", (message) => {
+                const response: BeamResult = JSON.parse(message.data)
+                if (response.task !== this.currentTask) return
+                let site: string = response.from.split(".")[1]
+                let status: Status = response.status
+                let body: SiteData = (status === "succeeded") ? JSON.parse(atob(response.body)) : null;
 
-                const changes = new Map<string, Site>();
-                beamResponseData.forEach((response: BeamResult) => {
-                    if (response.task !== this.currentTask) return;
-                    const site: string = response.from.split(".")[1];
-                    const status: Status = response.status;
-                    const body: SiteData =
-                        status === "succeeded"
-                            ? JSON.parse(atob(response.body))
-                            : null;
+                responseStore.update((store: ResponseStore): ResponseStore => {
+                    store.set(site, { status: status, data: body })
+                    return store;
+                })
+            })
 
-                    // if the site is already in the store and the status is claimed, don't update the store
-                    if (this.storeCache.get(site)?.status === status) return;
+            // read error events from beam
+            eventSource.addEventListener("error", (message) => {
+                console.log(`Beam returned error ${message}`)
+                eventSource.close()
+            })
 
-                    changes.set(site, { status: status, data: body });
-                });
-                if (changes.size > 0) {
-                    responseStore.update(
-                        (store: ResponseStore): ResponseStore => {
-                            changes.forEach((value, key) => {
-                                store.set(key, value);
-                            });
-                            return store;
-                        },
-                    );
-                }
-
-                responseCount = beamResponseData.length;
-                const realResponseCount = beamResponseData.filter(
-                    (response) => response.status !== "claimed",
-                ).length;
+            // event source in javascript throws an error then the event source is closed by backend
+            eventSource.onerror = () => {
```

Comment:
Maybe add a console.error here

Reply:
An error log is not the right choice here. Suprisingly, the eventSource.onerror event is thrown after the backend closes the EventStream, so actually after the beam proxy received answers from all sites. I added an info log instead.